cmd/bnsd/client/client.go

```diff
@@ -392,6 +394,33 @@ func walletKeyToAddr(key []byte) weave.Address {
 	return key[5:]
 }
 
+type CurrenciesResponse struct {
+	Height     int64
+	Currencies map[string]currency.TokenInfo
+}
+
+func (b *BnsClient) ListCurrencies() (*CurrenciesResponse, error) {
+	resp, err := b.AbciQuery("/tokens?prefix", nil)
+	if err != nil {
+		return nil, err
+	}
+	if len(resp.Models) == 0 { // empty list or nil
+		return nil, nil
```

Comment:
Method name suggests that a list or a collection is returned. Would it be better to change the notation to `ListCurrencies() (int, []currency.TokenInfo, error)` to avoid returning `nil, nil`? Using `nil` as a slice value is not an issue, while the `nil` structure pointer must be tested before using.

Reply:
easier to return an empty struct I suppose, but nil, nil could indeed be dangerous

Reply:
Other methods in this package return `nil, nil`, too but I will change this one.

Reply:
Well I don't suppose it's a good enough excuse.
In general I'd say if you return a nil value, then the error should be not nil.
Another great approach would be to return nil, errNotFound or errEmptyResult to be able to match/handle it better.